src/models/WidgetEventCapability.ts

```diff
@@ -96,11 +127,37 @@ export class WidgetEventCapability {
             // so we split on that. However, a # is also valid in either one of those so we
             // join accordingly.
             // Eg: `m.room.message##m.text` is "m.room.message" event with msgtype "#m.text".
+            const expectingKeyStr = eventSegment.startsWith("m.room.message#") || isState;
             let keyStr: string = null;
-            if (eventSegment.includes('#')) {
+            if (eventSegment.includes('#') && expectingKeyStr) {
+                // Dev note: regex is difficult to write, so instead the rules are manually written
```

Comment:
It's a bit sad we can't use an object and need to pack strings like this... 😓

Reply:
Yea... A future MSC should fix this.